src/App.tsx

```diff
@@ -15,48 +22,110 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+function getSortGoods(
+  goods: string[],
+  sortField: SortType,
+  reverseField: boolean,
+) {
+  const listOfGoods = [...goods];
+
+  if (sortField) {
+    listOfGoods.sort((good1, good2) => {
+      switch (sortField) {
+        case SortType.BY_ALPHABET:
+          return good1.localeCompare(good2);
+
+        case SortType.BY_LENGTH:
+          return good1.length - good2.length;
+
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (reverseField) {
+    listOfGoods.reverse();
+  }
+
+  return listOfGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortField, setSortField] = useState(SortType.BY_DEFAULT);
+  const [reverseField, setReverseField] = useState(false);
+
+  const visibleGoods = getSortGoods(
+    goodsFromServer,
+    sortField,
+    reverseField,
+  );
+
   return (
     <div className="section content">
       <div className="buttons">
         <button
           type="button"
-          className="button is-info is-light"
+          onClick={() => setSortField(SortType.BY_ALPHABET)}
+          className={
+            classNames(
+              'button is-info',
+              { 'is-light': sortField !== SortType.BY_ALPHABET },
+            )
+          }
         >
           Sort alphabetically
         </button>
 
         <button
           type="button"
-          className="button is-success is-light"
+          onClick={() => setSortField(SortType.BY_LENGTH)}
+          className={
+            classNames(
+              'button is-success',
+              { 'is-light': sortField !== SortType.BY_LENGTH },
+            )
+          }
         >
           Sort by length
         </button>
 
         <button
           type="button"
-          className="button is-warning is-light"
+          onClick={() => setReverseField(reverse => !reverse)}
```

Comment:
make function in component and pass reference here
```
const onReverse = () => setReverseField(reverse => !reverse)
```
```suggestion
          onClick={onReverse}
```